raft-tech/TANF-app
# Issue 738: Add Dropdown for Quarters Selection

### Summary of Changes
This PR adds the dropdown to select a `Quarter` and adds validation to the form so as to prevent searches without selecting the required options.

![2021-04-16 15 22 34](https://user-images.githubusercontent.com/12901172/115080055-1fde9880-9ed0-11eb-8880-cab7422a9c3b.gif)

_Addresses issue #738
_Acceptance criteria as stated in the issue_

**Tasks:**
- [x] Build Quarters dropdown with four options.
- [x] Add Dropdown to Reports page below the Years dropdown.
- [x] Update Redux  with a selectedQuarter state, and create related action(s).
- [x] Connect Dropdown to the Redux state.
- [x] Add appropriate UI and Redux tests.

### How to Test
_List the steps to test the PR_
1. Log in to the app with any user
2. Navigate to the reports tab
3. Select a Year and STT from their respective dropdown elements
4. Ensure that the Quarters dropdown is visible
5. Select any quarter from the dropdown
6. Ensure the selected quarter changes
7. Click Search and check that the UploadReport section is rendered with a visible header that includes `[Selected STT] - Fiscal Year [Selected Year] - [Selected Quarter]`
8. Refresh the page and reset the form
9. Without making any selections, click Search and make sure the error states are visible in each dropdown section
10. Make changes to any of the drop-down elements and make sure the errors disappear

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

	- [x] Dropdown allows user to select a Quarter on the Reports page.
	- [x] Users see error statements when clicking Search without selecting a Year and STT and Quarter

As facilitator/product manager, @kniz-raft will decide if ACs are met from Raft's perspective.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*


### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*

    + [See the full Accessibility Assessment plan here.](https://github.com/HHS/TANF-app/blob/1c12793a3c43612c8f6062c822edffc221f17ea8/docs/Technical-Documentation/how-government-will-test-a11y.md)

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

> NOTE: until we have a proper staging environment this may not be satisfiable prior to merging


### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*


### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

	- No issues found.
